modules/govuk_elasticsearch/manifests/init.pp

```diff
@@ -106,13 +121,18 @@
   }
 
   elasticsearch::instance { $::fqdn:
+    notify        => Exec['secure_es_yaml'],
     config        => $instance_config_real,
     datadir       => '/mnt/elasticsearch',
     init_defaults => {
       'ES_HEAP_SIZE' => $heap_size,
     },
   }
 
+  exec { 'secure_es_yaml':
+    command => "chmod 0640 /etc/elasticsearch/${::fqdn}/elasticsearch.yml",
+  }
```

Comment:
This is very noisy in the Puppet run output. On every run the elasticsearch module changes the mode from 640 to 644 and then this exec changes it back again.